### PR TITLE
chore: enforce conventional commits #40

### PR DESCRIPTION
**Pull request by SPlanzer**
Fixes: #  
#40 -  enforce conventional commits

### Change Description:
* conventional commit configuration 
* github action to run conventionalcommit validation

### Notes for Testing:
Github action will test commit 

#### Source Code Documentation Tasks:
- [x] README updated (where applicable)
- [x] CHANGELOG updated

#### User Documentation Tasks:
- [x] Confluence user guide updated (where applicable)

#### Testing Tasks:
- [x] Added tests that fail without this change
- [x] All tests are passing in development environment
- [x] Reviewers assigned

#### Pull Request Management: 
- [x] Linked to sub-task
- [x] Linked to the epic

### CODE REVIEW

**Comment by billgeo** [comment]
Not sure I'm the best to review this. Would suggest Ivan or Sandro.

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+name: Build
+on: [push]
+
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: ['3.6', '3.7']
```
Comment (strk) [inline .github/workflows/push.yml]: What's the point of linting with 2 different python version ?
Is it possible for a code to be found OK with 3.7 but dangerous/bogus with 3.6 ?

Comment (SPlanzer) [reply .github/workflows/push.yml]: Sorry @strk have only seen you have commented on this. 

This is following the recommendations of the [template-python-hello-world](https://github.com/linz/template-python-hello-world). This repo has been set up as a template for LINZ python projects. 

>What's the point of linting with 2 different python version ?
Is it possible for a code to be found OK with 3.7 but dangerous/bogus with 3.6 ?

Perhaps @dwsilk can comment?

Comment (SPlanzer) [reply .github/workflows/push.yml]: QGIS uses 3.6.8. TODO// Remove 3.7

**Pull request merged**